speaker/bgp_controller.go

```diff
@@ -83,48 +84,50 @@ newPeers:
 }
 
 // nodeHasHealthyEndpoint return true if this node has at least one healthy endpoint.
-func nodeHasHealthyEndpoint(eps *v1.Endpoints, node string) bool {
+func hasHealthyEndpoint(eps k8s.EpsOrSlices, filterNode func(*string) bool) bool {
 	ready := map[string]bool{}
-	for _, subset := range eps.Subsets {
-		for _, ep := range subset.Addresses {
-			if ep.NodeName == nil || *ep.NodeName != node {
-				continue
+	switch eps.Type {
+	case k8s.Eps:
+		for _, subset := range eps.EpVal.Subsets {
+			for _, ep := range subset.Addresses {
+				if filterNode(ep.NodeName) {
+					continue
+				}
+				if _, ok := ready[ep.IP]; !ok {
+					// Only set true if nothing else has expressed an
+					// opinion. This means that false will take precedence
+					// if there's any unready ports for a given endpoint.
+					ready[ep.IP] = true
+				}
 			}
-			if _, ok := ready[ep.IP]; !ok {
-				// Only set true if nothing else has expressed an
-				// opinion. This means that false will take precedence
-				// if there's any unready ports for a given endpoint.
-				ready[ep.IP] = true
+			for _, ep := range subset.NotReadyAddresses {
+				ready[ep.IP] = false
 			}
 		}
-		for _, ep := range subset.NotReadyAddresses {
-			ready[ep.IP] = false
-		}
-	}
-
-	for _, r := range ready {
-		if r {
-			// At least one fully healthy endpoint on this machine.
-			return true
-		}
-	}
-	return false
-}
+	case k8s.Slices:
+		for _, slice := range eps.SlicesVal {
+			for _, ep := range slice.Endpoints {
+				node := ep.Topology["kubernetes.io/hostname"]
```

Comment:
Topology is going to be deprecated, you have to use NodeName https://github.com/kubernetes/kubernetes/commit/9613f79adde3e0080fe25576cb37a74f25262848

Reply:
@aojea if I use NodeName, this will make this instance compatible with 1.22 only right?

Reply:
ups, wait, this is featuregated, you are right,

Reply:
so ... I think that you have to use topology meanwhile we are in discoveryv1beta1, and add a comment explaining the conversion that has to be done when it moves to discoveryv1 ...
how is metallb compatibility matrix with kubernetes? at some point when discoverv1beta1 is removed from the apiserver the project has to take a decision